### PR TITLE
Support dynamic (Jinja-generated) classes

### LINKED ISSUE #7: Dynamic css classes at render time
Hey, in my case there are a couple of colors i input in model fields interactively to include them in html later on. It looks like:

```ini
[fields.color_1]
label = Color
type = string
```

```html
<span class="text-[{{ this.color_1] }}"  # rendered class is like `text-[#735461]`
```

The problem is that `tailwindcss` can't find the final state of html classes BEFORE all the artifacts(html files) have been built. Manually i do for it:
1. `lektor build`
2. Configure `tailwindcss` to inspect `build` folder for built htmls.
3. `npx tailwindcss --input input.css --output build/static/style.css`
4. publish artifacts

Does the plugin take into account such the case? The result i got using plugin seems to be not contained the dynamic css classes.

### PR DESCRIPTION
**Pull request by mathrick**
Run tailwindcss on the build dir, rather than input templates, so that dynamically-generated values are properly seen by tailwind.

Fixes #7.

### CODE REVIEW

File: lektor_tailwind.py [hunk 1]
```diff
@@ -22,51 +22,46 @@ def __init__(self, env, id):
...
-                "-i",
-                self.input_css,
-                "-o",
-                os.path.join(output_path, self.css_path),
-                "--minify",
-            ],
+            self._get_tailwind_args(output_path, "--minify"),
             check=True,
-            cwd=self.env.root_path,
+            cwd=output_path,
```
Comment (frostming) [inline lektor_tailwind.py]: But this method is run `before_build`, I doubt if it works

Comment (mathrick) [reply lektor_tailwind.py]: You're right, fixed.

**Comment by mathrick** [comment]
I've removed the use of `tailwindcss -w`, since it never worked reliably, and it doesn't really make sense to use it in the first place. We already have a watcher built in.

**Comment by mathrick** [comment]
Is there anything else you'd like to see for this PR to be acceptable?

**Comment by frostming** [comment]
> Is there anything else you'd like to see for this PR to be acceptable?

Since you removed the tailwind live server, does it work in dev mode? Can it update the stylesheet on class changes? I just didn't find the time to verify it myself

**Comment by mathrick** [comment]
> Since you removed the tailwind live server, does it work in dev mode? Can it update the stylesheet on class changes? I just didn't find the time to verify it myself

Yes, I use it primarily in dev mode. It actually works *better* now, because the tailwind watcher no longer leaks memory, and it's more reliable. I believe the watcher would previously race lektor sometimes and not get everything updated. I've not had that happen since I removed `-w`, it's now very solid.

**Comment by frostming** [comment]
@mathrick Can you check the failing tests?

**Comment by mathrick** [comment]
@frostming will do

**Comment by mathrick** [comment]
@frostming: fixed, could you retry? 

I've been running into weird timeouts in `test_server_rebuilds_css_when_input_css_updated()` that I really can't see any reason for, and it shouldn't be any of my changes anyway. I had it pass once for no visible reason and then it went back to timing out, so I'm hoping it will just not do that when run through GitHub.

**Comment by mathrick** [comment]
This is really weird: for some reason, writing the sentinel value in `test_server_rebuilds_css_when_input_css_updated()` does not trigger a rebuild. But if I watch the test and manually change the file from another terminal before the timeout hits, the server happily picks up the change.

**Comment by mathrick** [comment]
To make things even weirder, this does *not* happen if I enter the debugger from within pytest. Which makes me think this is some odd artefact of how the server is run within the tests, and also makes it incredibly difficult to debug. Maybe something with buffering? But I'm not sure what would be buffered exactly that could cause that.

**Comment by mathrick** [comment]
Alright, this was extremely painful to debug, but I finally found the issue. To quote from the commit message:

> This was caused by the switch from having two watchers (lektor server and tailwindcss) to just the server. Because we made writes as soon as wait_for_server() returns, this would essentially catch the server in a short window before it was done returning from a build, and it wouldn't pick up the changes, so the rebuild would never happen. Previously this was not a concern, because the CSS was watched by one process, and other files by the other, and they delayed each other, so there was never a chance to write immediately after the rebuild was done. Now we sleep for a second before writing to prevent that.

> Note that this is purely an artefact of the testing process, and the fact we can do it within milliseconds. This is not possible for a human, so it doesn't come up in normal usage.

**Pull request merged**